doc/source/train/user-guides/hyperparameter-optimization.rst

```diff
@@ -3,127 +3,168 @@
 
-Basic usage
------------
+Set ``max_concurrent_trials`` to limit the number of Ray Train driver processes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Ray Train runs can only start when resources for all workers can be acquired at once.
+This means that multiple Tune trials spawning Train runs will be competing for the logical resources available in the Ray cluster.
 
-You can take an existing :class:`Trainer <ray.train.base_trainer.BaseTrainer>` and simply
-pass it into a :class:`~ray.tune.Tuner`.
+If there is a limiting cluster resource such as GPUs, then it won't be possible to run training for all hyperparameter configurations concurrently.
+Since the cluster only has enough resources for a handful of trials to run concurrently,
+set :class:`tune.TuneConfig(max_concurrent_trials) <ray.tune.TuneConfig>` on the Tuner to limit the number of “in-flight” Train runs so that no trial is being starved of resources.
 
-.. literalinclude:: ../doc_code/tuner.py
+.. literalinclude:: ../doc_code/train_tune_interop.py
     :language: python
-    :start-after: __basic_start__
-    :end-before: __basic_end__
+    :start-after: __max_concurrent_trials_start__
+    :end-before: __max_concurrent_trials_end__
 
 
+As a concrete example, consider a fixed sized cluster with 128 CPUs and 8 GPUs.
 
-How to configure a Tuner?
--------------------------
+* The ``Tuner(param_space)`` sweeps over 4 hyperparameter configurations with a grid search: ``param_space={“train_loop_config”: {“batch_size”: tune.grid_search([8, 16, 32, 64])}}``
+* Each Ray Train run is configured to train with 4 GPU workers: ``ScalingConfig(num_workers=4, use_gpu=True)``. Since there are only 8 GPUs, only 2 Train runs can acquire their full set of resources at a time.
+* However, since there are many CPUs available in the cluster, the 4 total Ray Tune trials (which default to requesting 1 CPU) can be launched immediately.
+  This results in 2 extra Ray Tune trial processes being launched, even though their inner Ray Train run just waits for resources until one of the other trials finishes.
+  This introduces some spammy log messages when Train waits for resources, and this could also lead to an excessive number of Ray Tune trial processes if the total number of hyperparameter configurations is large.
+* To fix this issue, set ``Tuner(tune_config=tune.TuneConfig(max_concurrent_trials=2))``. Now, only two Ray Tune trial processes will be running at a time.
+  This number can be calculated based on the limiting cluster resource and the amount of that resources required by each trial.
 
-There are two main configuration objects that can be passed into a Tuner: the :class:`TuneConfig <ray.tune.tune_config.TuneConfig>` and the :class:`RunConfig <ray.train.RunConfig>`.
 
-The :class:`TuneConfig <ray.tune.TuneConfig>` contains tuning specific settings, including:
+Advanced: Set Tune function resources
```

Comment:
non-blocking: we should standardize on "Tune function" vs. "Train driver function", so the terms in the quickstart match the terms used in the headers